Pass image_pull_policy in KubernetesPodOperator correctly

KevYuen opened the pull request:
image_pull_policy is not being passed into the V1Container in `KubernetesPodOperator`

This causes pods to not re-pull images even if the value is set.

KevYuen committed:
pass image_pull_policy to V1Container

image_pull_policy is not being passed into the V1Container in
KubernetesPodOperator.

KevYuen committed:
fix test for image_pull_policy in kubernetes provider

XD-DENG commented on tests/providers/cncf/kubernetes/operators/test_kubernetes_pod.py:
```diff
@@ -99,6 +99,33 @@ def test_image_pull_secrets_correctly_set(self, mock_client, monitor_mock, start
+    @mock.patch("airflow.kubernetes.pod_launcher.PodLauncher.start_pod")
+    @mock.patch("airflow.kubernetes.pod_launcher.PodLauncher.monitor_pod")
+    @mock.patch("airflow.kubernetes.kube_client.get_kube_client")
+    def test_image_pull_policy_correctly_set(self, mock_client, monitor_mock, start_mock):
```
Shall we have one more test case, where we don't explicitly specify `image_pull_policy` then assert it against default value `IfNotPresent `?

KevYuen committed:
add test for image_pull_policy not set

image_pull_policy should be IfNotPresent by default if
it's not set. The test ensure the correct value is passed
to the V1Container object.

XD-DENG reviewed (approved):
Thanks @KevYuen !

We can merge this after the CI succeeds. I will take care of it.

Thanks for the contribution!
